src/Chainweb/Pact/PactService/Checkpointer.hs

```diff
@@ -645,11 +645,19 @@ failOnTooLowRequestedHeight
     -> Maybe RewindLimit
     -> BlockHeader
     -> PactServiceM logger tbl ()
-failOnTooLowRequestedHeight parent (Just limit) lastHeader
-    | parentHeight + 1 + limitHeight < lastHeight = -- need to stick with addition because Word64
-        throwM $ RewindLimitExceeded limit parentHeight lastHeight parent
-  where
-    limitHeight = BlockHeight $ _rewindLimit limit
-    parentHeight = _blockHeight parent
-    lastHeight = _blockHeight lastHeader
+failOnTooLowRequestedHeight parent (Just limit) lastHeader = do
+  let limitHeight = BlockHeight $ _rewindLimit limit
+  let parentHeight = _blockHeight parent
+  let lastHeight = _blockHeight lastHeader
+  when (parentHeight + 1 + limitHeight < lastHeight) $ do -- need to stick with addition because Word64
+    throwM $ RewindLimitExceeded limit parentHeight lastHeight parent
+
+  cp <- getCheckpointer
+  mEarliestBlock <- liftIO $ _cpGetEarliestBlock cp
+  case mEarliestBlock of
+    Nothing -> error ""
```

Comment:
What should this say? When is this even possible?

Reply:
I think this is at genesis time, I'd just treat that the same as the height being 0.